invenio_app_ils/search_permissions.py

```diff
@@ -62,10 +62,15 @@ def _ils_search_factory(self, search, validator):
     :param search: Elastic search DSL search instance.
     :returns: Tuple with search instance and URL arguments.
     """
+
     def query_parser(qstr=None):
         """Default parser that uses the Q() from elasticsearch_dsl."""
         if qstr:
-            return Q('query_string', query=qstr)
+            return Q(
```

Comment:
I think that the idea is good, now we should maybe make it applicable to all type of records.

In `elasticsearch-py`, in the search class, we have:
```
class Meta:
    ...
    fields = ('*', )
```
See it [here](https://github.com/inveniosoftware/invenio-search/blob/master/invenio_search/api.py#L77).
What happens if we that in each Meta of each each search class in ILS?
Can you check if they are taken into account when performing the search?

Otherwise, we could try to change the DocumentSearch class in `documents/search.py` (and the same for other records):
```
DocumentSearch():
    ...
    _boosted_fields = [...]
```

and then here, in this generic query parser, if the `search` var is the class above, something like:

```
if ...
    return Q("query_string",
                    query=qstr,
                    fields=search._boosted_fields or search.fields or ["*"],  # whatever makes sense as or
return Q(fields=...as above...)
```

I would avoid the copy/paste of the entire search_factory as much as possible.